spec/tic_tac_toe_spec.rb

```diff
@@ -0,0 +1,231 @@
+require_relative '../lib/tic_tac_toe'
+
+describe '../lib/tic_tac_toe.rb' do
+  describe TicTacToe do
+    describe '#initialize' do
+      it 'assigns an instance variable @board to an array with 9 blank spaces " "' do
+        game = TicTacToe.new
+        expect(game.instance_variable_get(:@board)).to eq([' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' '])
+      end
+    end
+
+    describe '#input_to_index' do
+      it "accepts the user's input (a string) as an argument" do
+        game = TicTacToe.new
+        expect { game.input_to_index }.to raise_error(ArgumentError)
+      end
+
+      it "converts the user's input (a string) into an integer" do
+        game = TicTacToe.new
+        expect(game.input_to_index('1')).to be_an(Integer)
+      end
+
+      it "converts the user's input from the user-friendly format (on a 1-9 scale) to the array-friendly format (where the first index starts at 0)" do
+        game = TicTacToe.new
+        expect(game.input_to_index('1')).to eq(0)
+        expect(game.input_to_index('5')).to eq(4)
+      end
+    end
+
+    describe '#move' do
+      it 'allows "X" player in the top left and "O" in the middle' do
+        game = TicTacToe.new
+        game.move(0, 'X')
+        game.move(4, 'O')
+
+        board = game.instance_variable_get(:@board)
+
+        expect(board).to eq(['X', ' ', ' ', ' ', 'O', ' ', ' ', ' ', ' '])
+      end
+    end
+
+    describe '#position_taken?' do
+      it 'returns true/false based on whether the position on the board is already occupied' do
+        game = TicTacToe.new
+        board = ['X', ' ', ' ', ' ', ' ', ' ', ' ', ' ', 'O']
+        game.instance_variable_set(:@board, board)
+
+        index = 0
+        expect(game.position_taken?(index)).to be(true)
```

Comment:
- [ ] You should not expect multiple conditions from one _**it**_ block. Test all the positive and negative cases on different _**it**_ blocks. (This repeats sometimes in your code. Kindly fix them)
![image](https://user-images.githubusercontent.com/61479648/95194345-8bbaa880-07ab-11eb-9e71-bf29dfd62893.png)